### PR TITLE
[Tachyon-1237] Improved logging message formats in tachyon-servers

### PR DESCRIPTION
**Pull request by HorizonNet**
https://tachyon.atlassian.net/browse/TACHYON-1237

### CODE REVIEW

**Comment by yupeng9** [comment]
Thanks for the format improvement. It looks much better!

**Comment by yupeng9** [comment]
LGTM. Please resolve the conflicts.

File: servers/src/main/java/tachyon/master/MasterBase.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ public void processJournalCheckpoint(JournalInputStream inputStream) throws IOEx
   @Override
   public void start(boolean isLeader) throws IOException {
     mIsLeader = isLeader;
-    LOG.info(getServiceName() + ": Starting " + (mIsLeader ? "leader" : "standby") + " master.");
+    LOG.info("{}: Starting " + (mIsLeader ? "leader" : "standby") + " master.", getServiceName());
```
Comment (apc999) [inline servers/src/main/java/tachyon/master/MasterBase.java]: still having string concatenation here??
` ... + (mIsLeader ? "leader" : "standby") + ...`

File: servers/src/main/java/tachyon/master/MasterBase.java [hunk 2]
```diff
@@ -142,7 +142,7 @@ public void start(boolean isLeader) throws IOException {
 
   @Override
   public void stop() throws IOException {
-    LOG.info(getServiceName() + ": Stopping " + (mIsLeader ? "leader" : "standby") + " master.");
+    LOG.info("{}: Stopping " + (mIsLeader ? "leader" : "standby") + " master.", getServiceName());
```
Comment (apc999) [inline servers/src/main/java/tachyon/master/MasterBase.java]: same comment as above?

File: servers/src/main/java/tachyon/master/file/meta/InodeTree.java [hunk 3]
```diff
@@ -222,10 +222,10 @@ public CreatePathResult createPath(TachyonURI path, CreatePathOptions options)
       // exception here. Otherwise we add the remaining path components to the list of components
       // to create.
       if (!options.isRecursive()) {
-        final String msg = "File " + path + " creation failed. Component "
-            + traversalResult.getNonexistentPathIndex() + "("
-            + parentPath[traversalResult.getNonexistentPathIndex()] + ") does not exist";
-        LOG.info("InvalidPathException: " + msg);
+        final String msg = String.format("File %s creation failed. Component %d(%s) does not exist",
```
Comment (apc999) [inline servers/src/main/java/tachyon/master/file/meta/InodeTree.java]: StringBuilder might be a better choice here:
http://stackoverflow.com/questions/513600/should-i-use-javas-string-format-if-performance-is-important

File: servers/src/main/java/tachyon/worker/block/TieredBlockStore.java [hunk 4]
```diff
@@ -574,13 +574,13 @@ private TempBlockMeta createBlockMetaInternal(long sessionId, long blockId,
...
-        LOG.error("Unexpected failure: " + initialBlockSize + " bytes allocated at " + location
-            + " by allocator, but addTempBlockMeta failed");
+        LOG.error("Unexpected failure: {} bytes allocated at {} by allocator, "
+            + "but addTempBlockMeta failed", initialBlockSize, location);
         throw Throwables.propagate(ose);
       } catch (BlockAlreadyExistsException aee) {
         // If we reach here, allocator is not working properly
-        LOG.error("Unexpected failure: " + initialBlockSize + " bytes allocated at " + location
-            + " by allocator, but addTempBlockMeta failed");
+        LOG.error("Unexpected failure: %d bytes allocated at {} by allocator, "
```
Comment (apc999) [inline servers/src/main/java/tachyon/worker/block/TieredBlockStore.java]: is it correct to have `%d` here? I thought slf4j doesn't know `%d` but only `{}`?

File: servers/src/main/java/tachyon/worker/block/meta/StorageDir.java [hunk 5]
```diff
@@ -124,7 +124,7 @@ private void initializeMeta() throws BlockAlreadyExistsException, IOException,
           // TODO(calvin): Resolve this conflict in class names.
           org.apache.commons.io.FileUtils.deleteDirectory(path);
         } catch (IOException ioe) {
-          LOG.error("can not delete directory {}: {}", path.getAbsolutePath(), ioe);
+          LOG.error(String.format("can not delete directory %s", path.getAbsolutePath()), ioe);
```
Comment (apc999) [inline servers/src/main/java/tachyon/worker/block/meta/StorageDir.java]: for a single param log, you can achieve the same result in a cleaner way using 
`LOG.error("can not delete directory {}", path.getAbsolutePath(), ioe);`
which avoids using `String.format`.
This is supported after slf4j 1.7 
http://www.slf4j.org/faq.html#paramException

**Comment by apc999** [comment]
Thanks @HorizonNet , this is a nice cleanup and really improves readability. I have left a few comments or questions. Could you take a look,and also merge with latest master to resolve conflicts?  Thanks

**Comment by HorizonNet** [comment]
@apc999 Thanks for the comments. I updated the PR accordingly. Let me know what you're thinking.

File: servers/src/main/java/tachyon/master/file/meta/InodeTree.java [hunk 6]
```diff
@@ -222,10 +222,16 @@ public CreatePathResult createPath(TachyonURI path, CreatePathOptions options)
       // exception here. Otherwise we add the remaining path components to the list of components
       // to create.
       if (!options.isRecursive()) {
-        final String msg = "File " + path + " creation failed. Component "
-            + traversalResult.getNonexistentPathIndex() + "("
-            + parentPath[traversalResult.getNonexistentPathIndex()] + ") does not exist";
-        LOG.info("InvalidPathException: " + msg);
+        StringBuilder builder = new StringBuilder();
```
Comment (apc999) [inline servers/src/main/java/tachyon/master/file/meta/InodeTree.java]: this block of code can be consolidated: 
```java
StringBuilder builder = new StringBuilder().append("File ").append(path)....toString();
```

**Comment by apc999** [comment]
Thanks @HorizonNet , I have my final comment left. After this, I will LGTM this PR. Thanks a lot!

**Comment by HorizonNet** [comment]
@apc999 Thanks for the comment! I included it in my latest commit.

**Comment by apc999** [comment]
+1 LGTM

**Pull request merged**